tests/soak-tests/src/test/java/org/apache/activemq/artemis/tests/soak/interruptlm/LargeMessageInterruptTest.java

```diff
@@ -0,0 +1,244 @@
+            for (int i = 0; i < 1000; i++) {
+               producer.send(session.createTextMessage("forcePage"));
+            }
+            session.commit();
+         }
+      }
+
+      for (int i = 0; i < SENDING_THREADS; i++) {
+         executorService.execute(() -> {
+            int numberOfMessages = 0;
+            try {
+               Connection connection = factory.createConnection();
+               Session session = connection.createSession(tx, tx ? Session.SESSION_TRANSACTED : Session.AUTO_ACKNOWLEDGE);
+               MessageProducer producer = session.createProducer(session.createQueue(queueName));
+
+               startFlag.await(10, TimeUnit.SECONDS);
+               while (numberOfMessages < NUMBER_OF_MESSAGES) {
+                  try {
+                     producer.send(session.createTextMessage(largebody));
+                     if (tx) {
+                        session.commit();
+                     }
+                     produced.incrementAndGet();
+                     killAt.countDown();
+                     if (numberOfMessages++ % 10 == 0) {
+                        logger.info("Sent {}", numberOfMessages);
+                     }
+                  } catch (Exception e) {
+                     logger.warn(e.getMessage(), e);
+
+                     logger.warn(e.getMessage(), e);
+                     try {
+                        connection.close();
+                     } catch (Throwable ignored) {
+                     }
+
+                     for (int retryNumber = 0; retryNumber < 100; retryNumber++) {
+                        try {
+                           Connection ctest = factory.createConnection();
+                           ctest.close();
+                           break;
+                        } catch (Throwable retry) {
+                           Thread.sleep(100);
+                        }
+                     }
+
+                     connection = factory.createConnection();
+                     session = connection.createSession(tx, tx ? Session.SESSION_TRANSACTED : Session.AUTO_ACKNOWLEDGE);
+                     producer = session.createProducer(session.createQueue(queueName));
+                     connection.start();
+
+                  }
+               }
+            } catch (Exception e) {
+               logger.warn("Error getting the initial connection", e);
+               errors.incrementAndGet();
+            }
+
+            logger.info("Done sending");
+            done.countDown();
+         });
+      }
+
+      Assert.assertTrue(killAt.await(60, TimeUnit.SECONDS));
+      serverProcess.destroyForcibly();
+      serverProcess = startServer(SERVER_NAME_0, 0, 0);
+      QueueControl queueControl = getQueueControl(liveURI, liveNameBuilder, queueName, queueName, RoutingType.ANYCAST, 5000);
+
+      Assert.assertTrue(done.await(60, TimeUnit.SECONDS));
+      Assert.assertEquals(0, errors.get());
+
+      long numberOfMessages = queueControl.getMessageCount();
```

Comment:
Its not clear to me why this seems to go to the effort of starting the broker without waiting, spinning in a loop trying to connect to management to get a QueueControl (and potentially waiting 500ms at a time, and spewing tracktrace, while doing it), only to not use the QueueControl until after it is done waiting for sending to complete?

Reply:
I am actually using it to get the numberOfMessages.

Reply:
Sure, I didnt say it was unused, only that it goes to the effort of spinning and potentially spewing stacktraces by connecting before the broker is ready...only to wait for the sending to complete before then using it. Why not create it after the wait for send completion when we know the broker is ready?

Reply:
@gemmellr OH.. I see what you mean...


I was going to place either that loop, or a connection loop over JMS... to wait the server to start.
Since the threads already have a retry, I didn't consider to use them as the retry mechanism, which works perfectly. Kind of obvious but I was too stuck in my line of though.. thanks.


I'm moving down the queueControl creation by 2 lines.